### DISCUSSION TITLE
Migrate a single repo from Github enterprise

### DISCUSSION THREAD

**Discussion by ddevarap8** [community/community#116760] [Enterprise]
### Select Topic Area

Question

### Body

We are looking for migrating a single repo from Github enterprise to different Github enterprise server. We are expecting to migrate with **Pull** **Requests and source code, tags.**

Do we have any tool or scripts to handle this scenario ?

**Comment by santoshvandari** [comment]
You can use the GitHub CLI to migrate repositories. The GitHub CLI simplifies the migration process and is recommended.

**Comment by giovannivicentin** [comment]
Hi ddevarap8,

Migrating a repository from one GitHub Enterprise server to another, including all its associated metadata like pull requests, source code, and tags, requires a comprehensive approach to ensure that no critical information is lost in the transition. While GitHub doesn't offer a built-in tool specifically for cross-server migrations, there are several strategies and third-party tools you can consider to facilitate this process:

1. **GitHub's Official Import/Export Feature**:
   - GitHub Enterprise has a feature for [migrating repositories](https://docs.github.com/en/enterprise-server@2.22/admin/installation/migrating-from-githubcom-to-github-enterprise-server) that includes issues, pull requests, and other essential data. However, this feature is typically used for migrating from GitHub.com to GitHub Enterprise Server and might have some limitations when moving between Enterprise instances. It's worth checking with GitHub's support or your GitHub Enterprise documentation to see if this can be adapted to your needs.

2. **Third-party Tools**:
   - Tools like [gh-ost](https://github.com/github/gh-ost) and [GitHub Migration Tool](https://github.com/github/gh-migration) can assist in migrating repository data. These tools are often open-source and can be adapted to your specific requirements, but they might require some technical expertise to use effectively.

3. **Custom Scripts**:
   - Writing custom scripts using GitHub's API is another way to migrate repository data. The GitHub API allows you to extract and push a wide range of repository data, including pull requests, commits, and tags. However, this approach requires a good understanding of the GitHub API and scripting (usually with Python or a similar language). Example scripts and libraries can often be found on GitHub or in the GitHub API documentation.

4. **Considerations for Pull Requests**:
   - Migrating pull requests can be particularly challenging because they involve not just the code changes but also comments, review statuses, and more. When using scripts or tools, ensure they are capable of handling the complex data associated with pull requests. You may need to accept that some metadata (like review statuses) could be lost or need to be manually recreated.

5. **Testing the Migration**:
   - Before performing the full migration, it's crucial to conduct a test run with a copy of your repository. This allows you to identify any issues or data loss in a controlled environment, ensuring the actual migration goes as smoothly as possible.

6. **Consult GitHub Enterprise Support**:
   - Given the complexity of such a migration, reaching out to GitHub Enterprise Support is advisable. They might offer guidance, best practices, and insights specific to your version of GitHub Enterprise.

7. **Documentation and Communication**:
   - Document the migration process and communicate with your team about any potential access changes or workflow disruptions during the migration. Ensure that everyone involved is aware of the timeline and any actions they may need to take.

If you're looking for more detailed guidance or if you have specific technical questions about the migration process, feel free to share more details or ask follow-up questions.

Best of luck with your migration project!
